spring-integration-mail/src/main/java/org/springframework/integration/mail/dsl/Mail.java

```diff
@@ -0,0 +1,152 @@
+		return pop3InboundAdapter(host, -1, username, password);
+	}
+
+	/**
+	 * A {@link Pop3MailInboundChannelAdapterSpec} factory based on the provided host,
+	 * port, user, and password.
+	 * @param host the host.
+	 * @param port the port.
+	 * @param username the user.
+	 * @param password the password.
+	 * @return the {@link Pop3MailInboundChannelAdapterSpec} instance.
+	 */
+	public static Pop3MailInboundChannelAdapterSpec pop3InboundAdapter(String host, int port, String username,
+			String password) {
+		return new Pop3MailInboundChannelAdapterSpec(host, port, username, password);
+	}
+
+	/**
+	 * An {@link ImapMailInboundChannelAdapterSpec} factory using a default {@link ImapMailReceiver}.
+	 * @return the {@link ImapMailInboundChannelAdapterSpec} instance.
+	 */
+	public static ImapMailInboundChannelAdapterSpec imapInboundAdapter() {
+		return new ImapMailInboundChannelAdapterSpec();
+	}
+
+	/**
+	 * An {@link ImapMailInboundChannelAdapterSpec} factory based on the provided {@link ImapMailReceiver}.
+	 * @param imapMailReceiver the {@link ImapMailReceiver} to use.
+	 * @return the {@link ImapMailInboundChannelAdapterSpec} instance.
+	 */
+	public static ImapMailInboundChannelAdapterSpec imapInboundAdapter(ImapMailReceiver imapMailReceiver) {
+		return new ImapMailInboundChannelAdapterSpec(imapMailReceiver);
+	}
+
+	/**
+	 * A {@link ImapMailInboundChannelAdapterSpec} factory based on the provided url.
+	 * @param url the imap url
+	 * @return the {@link ImapMailInboundChannelAdapterSpec} instance.
+	 */
+	public static ImapMailInboundChannelAdapterSpec imapInboundAdapter(String url) {
+		return new ImapMailInboundChannelAdapterSpec(url);
+	}
+
+	/**
+	 * An {@link ImapIdleChannelAdapterSpec} factory using a default {@link ImapMailReceiver}.
+	 * @return the {@link ImapIdleChannelAdapterSpec} instance.
+	 */
+	public static ImapIdleChannelAdapterSpec imapIdleAdapter() {
+		return new ImapIdleChannelAdapterSpec(new ImapMailReceiver());
+	}
+
+	/**
+	 * A {@link ImapIdleChannelAdapterSpec} factory based on the provided url.
+	 * @param url the imap url
+	 * @return the {@link ImapIdleChannelAdapterSpec} instance.
+	 */
+	public static ImapIdleChannelAdapterSpec imapIdleAdapter(String url) {
+		return new ImapIdleChannelAdapterSpec(new ImapMailReceiver(url));
+	}
+
+	/**
+	 * An {@link ImapIdleChannelAdapterSpec} factory based on the provided {@link ImapMailReceiver}.
+	 * @param imapMailReceiver the {@link ImapMailReceiver} to use.
+	 * @return the {@link ImapIdleChannelAdapterSpec} instance.
+	 */
+	public static ImapIdleChannelAdapterSpec imapIdleAdapter(ImapMailReceiver imapMailReceiver) {
+		return new ImapIdleChannelAdapterSpec(imapMailReceiver, true);
+	}
+
+	/**
+	 * A {@link MailHeadersBuilder} factory.
+	 * @return the factory.
+	 */
+	public static MailHeadersBuilder headers() {
+		return new MailHeadersBuilder();
+	}
+
+	private Mail() {
+	}
```

Comment:
`super()` for Sonar.
And there is `MailToStringTransformer` to add to this Factory.